chain/bitcoind_conn.go

```diff
@@ -370,11 +447,149 @@ func (c *BitcoindConn) blockEventHandler() {
 	}
 }
 
+// blockEventHandlerRPC is a goroutine that uses the rpc client to check if we
+// have a new block every so often.
+func (c *BitcoindConn) blockEventHandlerRPC() {
+	defer c.wg.Done()
+
+	log.Info("Started polling for new bitcoind blocks via RPC.")
+	ticker := time.NewTicker(c.cfg.PollBlockTimer)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			// Every so many seconds, we poll to see if there's a new
+			// block.
+			newBlockHash, err := c.client.GetBestBlockHash()
+			if err != nil || newBlockHash == nil {
+				log.Errorf("Unable to retrieve best block "+
+					"hash: %v", err)
+				continue
+			}
+			newBlockHeader, err := c.client.GetBlockHeaderVerbose(
+				newBlockHash,
+			)
+			if err != nil {
+				log.Errorf("Unable to retrieve block height: "+
+					"%v", err)
+				continue
+			}
+
+			c.rescanClientsMtx.Lock()
+			if c.rescanClients == nil || len(c.rescanClients) < 1 {
+				log.Errorf("No client added yet")
+				c.rescanClientsMtx.Unlock()
```

Comment:
I dont think it is an error if there are no clients. Also, we still want to update our local mempool view even if there are no clients.